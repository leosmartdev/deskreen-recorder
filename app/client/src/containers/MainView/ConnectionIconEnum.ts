enum ConnectionIcon {
  FEED = 'feed',
  FEED_SUBSCRIBED = 'feed-subscribed'
}

export default ConnectionIcon;
