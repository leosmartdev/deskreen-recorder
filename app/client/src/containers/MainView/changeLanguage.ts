import i18n from "../../config/i18n";

export default (lng: string) => {
  i18n.changeLanguage(lng);
};
