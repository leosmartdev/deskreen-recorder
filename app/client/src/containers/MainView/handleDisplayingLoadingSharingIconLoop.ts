import LoadingSharingIconEnum from './LoadingSharingIconEnum';

export default (params: handleDisplayingLoadingSharingIconLoopParams) => {
  const {
    promptStep,
    url,
    setIsShownLoadingSharingIcon,
    loadingSharingIconType,
    isShownLoadingSharingIcon,
    setLoadingSharingIconType,
  } = params;
  return () => {
    let interval: NodeJS.Timeout;
    if (promptStep === 3 && url === undefined) {
      setIsShownLoadingSharingIcon(true);

      let currentIcon = loadingSharingIconType;
      let isShownIcon = isShownLoadingSharingIcon;
      let isShownWithFadingUIEffect = false;
      interval = setInterval(() => {
        isShownIcon = !isShownIcon;
        setIsShownLoadingSharingIcon(isShownIcon);
        if (isShownWithFadingUIEffect) {
          currentIcon =
            currentIcon === LoadingSharingIconEnum.DESKTOP
              ? LoadingSharingIconEnum.APPLICATION
              : LoadingSharingIconEnum.DESKTOP;
          setLoadingSharingIconType(currentIcon);
          isShownWithFadingUIEffect = false;
        } else {
          isShownWithFadingUIEffect = true;
        }
      }, 1500);
    }

    return () => {
      clearInterval(interval);
    };
  };
};
