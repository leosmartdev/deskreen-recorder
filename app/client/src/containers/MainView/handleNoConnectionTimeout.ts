import { DUMMY_MY_DEVICE_DETAILS } from "../../constants/appConstants";

export default (
  myDeviceDetails: DeviceDetails,
  setIsErrorDialogOpen: (_: boolean) => void
) => {
  return () => {
    const timeout = setTimeout(() => {
      if (myDeviceDetails === DUMMY_MY_DEVICE_DETAILS) {
        setIsErrorDialogOpen(true);
      }
    }, 10000);
    return () => {
      clearTimeout(timeout);
    };
  };
};
