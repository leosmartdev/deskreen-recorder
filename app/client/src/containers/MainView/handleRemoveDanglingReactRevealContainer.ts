export default (
  url: undefined | MediaStream
) => {
  return () => {
    if (url !== undefined) {
      setTimeout(() => {
        // @ts-ignore
        document.querySelector('.container > .react-reveal').style.display =
          'none';
      }, 1000);
    }
  };
};
