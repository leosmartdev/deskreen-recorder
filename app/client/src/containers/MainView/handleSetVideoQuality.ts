import PeerConnection from "../../features/PeerConnection";
import { VideoQuality } from "../../features/VideoAutoQualityOptimizer/VideoQualityEnum";

export default (
  videoQuality: VideoQuality,
  peer: PeerConnection | undefined
) => {
  return () => {
    if (!peer) return;
    if (!peer.isStreamStarted) return;
    peer.setVideoQuality(videoQuality);
  };
};
