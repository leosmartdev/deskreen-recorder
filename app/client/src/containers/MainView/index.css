@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }

  #pulsing-circle-1 {
    animation: pulse1 infinite 6s linear;
  }

  #pulsing-circle-2 {
    animation: pulse2 infinite 6s linear;
  }

  .pulse-3-once {
    animation: pulse3twice 2 750ms linear;
  }
}

@keyframes pulse1 {
	0% {
		transform: scale(0.95);
		box-shadow: 0 0 0 0 rgba(72, 175, 240, 0.7);
	}

	80% {
		transform: scale(1);
		box-shadow: 0 0 0 15px rgba(72, 175, 240, 0.4);
	}

	100% {
		transform: scale(0.95);
		box-shadow: 0 0 0 0 rgba(72, 175, 240, 0);
	}
}

@keyframes pulse2 {
	100% {
		transform: scale(0.95);
		box-shadow: 0 0 0 0 rgba(72, 175, 240, 0);
	}

	80% {
		transform: scale(1);
		box-shadow: 0 0 0 33px rgba(72, 175, 240, 0.3);
	}

	0% {
		transform: scale(0.95);
		box-shadow: 0 0 0 0 rgba(72, 175, 240, 1);
	}
}

@keyframes pulse3twice {
  0% {
		box-shadow: 0 0 0 0 rgba(21, 179, 113, 0.7);
  }

  50% {
    box-shadow: 0 0 0 30px rgba(21, 179, 113, 0.3);
  }

  100% {
    box-shadow: 0 0 0 0 rgba(21, 179, 113, 0);
  }
}


.container > .react-reveal {
  overflow: hidden;
}
