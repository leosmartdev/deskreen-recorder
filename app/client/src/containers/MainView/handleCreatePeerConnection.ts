import PeerConnection from '../../features/PeerConnection';
import PeerConnectionUIHandler from '../../features/PeerConnection/PeerConnectionUIHandler';
import getRoomIDOfCurrentBrowserWindow from '../../utils/getRoomIDOfCurrentBrowserWindow';
import Crypto from '../../utils/crypto';
import VideoAutoQualityOptimizer from '../../features/VideoAutoQualityOptimizer';
import changeLanguage from './changeLanguage';
import ConnectionIcon from './ConnectionIconEnum';

export default (params: CreatePeerConnectionUseEffectParams) => {
  const {
    isDarkTheme,
    peer,
    setIsDarkThemeHook,
    setMyDeviceDetails,
    setConnectionIconType,
    setIsShownTextPrompt,
    setPromptStep,
    setScreenSharingSourceType,
    setDialogErrorMessage,
    setIsErrorDialogOpen,
    setUrl,
    setPeer,
  } = params;

  return () => {
    if (!peer) {
      const UIHandler = new PeerConnectionUIHandler(
        isDarkTheme,
        setMyDeviceDetails,
        () => {
          setConnectionIconType(ConnectionIcon.FEED_SUBSCRIBED);

          setIsShownTextPrompt(false);
          setIsShownTextPrompt(true);
          setPromptStep(2);

          setTimeout(() => {
            setIsShownTextPrompt(false);
            setIsShownTextPrompt(true);
            setPromptStep(3);
          }, 2000);
        },
        setScreenSharingSourceType,
        setIsDarkThemeHook,
        changeLanguage,
        setDialogErrorMessage,
        setIsErrorDialogOpen
      );

      const _peer = new PeerConnection(
        getRoomIDOfCurrentBrowserWindow(),
        setUrl,
        new Crypto(),
        new VideoAutoQualityOptimizer(),
        UIHandler
      );

      setPeer(_peer);

      setTimeout(() => {
        setIsShownTextPrompt(true);
      }, 100);
    }
  };
};
