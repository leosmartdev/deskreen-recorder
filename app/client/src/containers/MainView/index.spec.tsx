import React, { Suspense } from 'react';

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import MainView from '.';
import { I18nextProvider } from 'react-i18next';

import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

// Mock react-i18next
i18n.use(initReactI18next).init({
  lng: 'en',
  fallbackLng: 'en',
  ns: ['common'],
  defaultNS: 'common',
  resources: {
    en: {
      common: {},
    },
  },
  debug: false,
});

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

it('should match exact snapshot', () => {
  const subject = shallow(
    <Suspense fallback={<div>loading</div>}>
      <I18nextProvider i18n={i18n}>
        <MainView />
      </I18nextProvider>
    </Suspense>
  );
  expect(subject).toMatchSnapshot();
});
