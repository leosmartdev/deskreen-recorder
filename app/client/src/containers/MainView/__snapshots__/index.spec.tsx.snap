// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match exact snapshot 1`] = `ShallowWrapper {}`;
